client/src/featureform/register.py

```diff
@@ -3947,6 +4010,194 @@ def search(self, raw_query, local=False):
             return search(processed_query, self._host)
 
 
+class ColumnResource:
```

Comment:
You're effectively creating an abstract class here right? You can force the inability to instantiate with this -> https://www.geeksforgeeks.org/abstract-classes-in-python/